lxc/config.go

```diff
@@ -874,3 +878,428 @@ func (c *cmdConfigUnset) Run(cmd *cobra.Command, args []string) error {
+func (c *cmdConfigUefiGet) Run(cmd *cobra.Command, args []string) error {
+	// Quick checks.
+	exit, err := c.global.CheckArgs(cmd, args, 2, 2)
+	if exit {
+		return err
+	}
+
+	// Parse remote
+	remote := args[0]
+	resources, err := c.global.ParseServers(remote)
+	if err != nil {
+		return err
+	}
+
+	resource := resources[0]
+	if resource.name == "" {
+		return fmt.Errorf(i18n.G("Instance name must be specified"))
+	}
+
+	// Get the EFI variable
+	resp, _, err := resource.server.GetInstanceEFIVars(resource.name)
+	if err != nil {
+		return err
+	}
+
+	efiVariable, ok := resp.Variables[args[len(args)-1]]
+	if !ok {
+		return fmt.Errorf(i18n.G("Requested EFI variable does not exist"))
+	}
+
+	fmt.Println(efiVariable.Data)
+
+	return nil
+}
+
+// Set.
+type cmdConfigUefiSet struct {
+	global     *cmdGlobal
+	configUefi *cmdConfigUefi
+}
+
+// Command creates a new Cobra command to set virtual machine instance UEFI variables.
+func (c *cmdConfigUefiSet) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = usage("set", i18n.G("[<remote>:]<instance> <key>=<value>..."))
+	cmd.Short = i18n.G("Set EFI variables for instance")
+	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
+		`Set EFI variables for instance`))
+	cmd.Example = cli.FormatSection("", i18n.G(
+		`lxc config uefi set [<remote>:]<instance> testvar-9073e4e0-60ec-4b6e-9903-4c223c260f3c=aabb
+    Will set a EFI variable with name "testvar", GUID 9073e4e0-60ec-4b6e-9903-4c223c260f3c and value "aabb" (HEX-encoded) for the instance.`))
+
+	cmd.RunE = c.Run
+
+	return cmd
+}
+
+// Run executes the "set" command, updating virtual machine instance UEFI variables based on provided arguments.
+func (c *cmdConfigUefiSet) Run(cmd *cobra.Command, args []string) error {
+	// Quick checks.
+	exit, err := c.global.CheckArgs(cmd, args, 2, -1)
+	if exit {
+		return err
+	}
+
+	// Parse remote
+	remote := args[0]
+	resources, err := c.global.ParseServers(remote)
+	if err != nil {
+		return err
+	}
+
+	resource := resources[0]
+	if resource.name == "" {
+		return fmt.Errorf(i18n.G("Instance name must be specified"))
+	}
+
+	// Set the config keys
+	keys, err := getConfig(args[1:]...)
+	if err != nil {
+		return err
+	}
+
+	instEFI, etag, err := resource.server.GetInstanceEFIVars(resource.name)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range keys {
+		if cmd.Name() == "unset" {
+			_, ok := instEFI.Variables[k]
+			if !ok {
+				return fmt.Errorf(i18n.G("Can't unset key '%s', it's not currently set"), k)
+			}
+
+			delete(instEFI.Variables, k)
+		} else {
+			// Initialize EFI variable attributes with the default value
+			_, ok := instEFI.Variables[k]
+			if !ok {
+				instEFI.Variables[k] = &api.InstanceEFIVariable{
+					// EFI_VARIABLE_NON_VOLATILE | EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS
+					Attr: 7,
+				}
+			}
+
+			instEFI.Variables[k].Data = v
+		}
+	}
+
+	err = instEFI.CheckValid()
```

Comment:
Lets let the server do the validation - in case we change the validation in the future and an older/newer client is talking to a different version of the LXD server.

Reply:
Fixed, thanks!